src/Compositor.cpp

```diff
@@ -2297,6 +2297,11 @@ void CCompositor::setWindowFullscreen(CWindow* pWindow, bool on, eFullscreenMode
         return;
     }
 
+    if (!pWindow->m_bIsFullscreen && !on) {
```

Comment:
why not just `if (fullscreen == on)`

(there is a very similar check for the opposite case a few lines down, but we can avoid a few calcs in that cdase)

Reply:
Well since there are actually two kinds of fullscreen, I thought there could be a scenario where someone might wanna call it with on = true to change it from fs full to maximized, or vice versa. However, in the current code it doesn't look like it would make a difference indeed. But I can change this if you want, doesn't hurt though imo

Reply:
that's a job for a different MR. I'd do `==` in this, and eventually build upon later

Reply:
fair enough, changed